hedera-node/infrastructure/docker/containers/production-next/consensus-node/entrypoint.sh

```
@@ -100,6 +100,10 @@ if [[ "${JCP_OVERRIDDEN}" != true && "${JAVA_MAIN_CLASS}" == "com.swirlds.platfo
   JAVA_CLASS_PATH="data/lib/*"
 fi
 
+# Setup Consensus Node Arguments
+CONSENSUS_NODE_ARGS=""
+[[ -n "${CONSENSUS_NODE_ID}" && "${CONSENSUS_NODE_ID}" -ge 0 ]] && CONSENSUS_NODE_ARGS="-local ${CONSENSUS_NODE_ID}"
```

is CONSENSUS_NODE_ID is not specified or less than 0 should this be treated as an error condition and not startup?

---

@dalvizu It ultimately will be at the NMT level. We could add a failure condition here; however, these images are also used for older software releases which do not have the `-local` option. 

We made it optional here for backwards compatibility with NMT 1.2.9 and `0.58.x` releases.

---

Ultimately, the `ServicesMain` entrypoint should fail fast if the required nodeId is not specified. 

@edward-swirldslabs Does the `ServicesMain` entrypoint fail if no environment variable and no `-local` switch is supplied?

---

@nathanklick Yes.  The call to `ensureSingleNode` makes sure there is no more and no less than 1 node being started.  

If no -local switch is supplied, then we're relying on the environment variable to determine the node to start.  If no environment variable is set, then the node will fail to start (from ServicesMain) because its singular identity was not specified.

---

@dalvizu Does this satisfy your concerns?

---

The commandline and environmment variables are parsed into `NodeId` constructs that wrap the long value.  If the long value provided is negative, the `NodeId` will fail with a java illegal argument exception.

---

@dalvizu  I am resolving this comment since there has been no reply